addons/purchase_stock/tests/test_reordering_rule.py

```diff
@@ -1357,3 +1357,68 @@ def test_reordering_rule_multicurrency(self):
             [("product_id", "=", self.product_01.id)])
         self.assertTrue(po_line)
         self.assertEqual(po_line.order_id.currency_id, foreign_currency)
+
+    def test_intercompany_reordering_rules(self):
+        """
+        Have 2 companies, create a procurment to fulfil a demand in COMP1 using custom route
+        with 2 rules: an intercompany transit from COMP2 to COMP1 and a buy rule linked to COMP2.
+
+        Check that the purchase order is created in COMP2, using its set of supplier.
+        """
+        company_a, company_b = self.env['res.company'].create([
+            {'name': 'Company A'},
+            {'name': 'Company B'},
+        ])
+        warehouse_a, warehouse_b = self.env['stock.warehouse'].search([('company_id', 'in', [company_a.id, company_b.id])], limit=2).sorted('company_id')
+        route_a_resupply_from_intercomp, route_b_ressuply_intercomp = self.env['stock.route'].create([
```

Comment:
Nitpick, but you could create a single route that has no company and set the company_id on each rule instead to save a few lines :slightly_smiling_face: